cmd/helm/completion.go

```diff
@@ -74,173 +74,23 @@ func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
-			# the value to print is -n, which would be interpreted
-			# as a flag to echo
-			printf "%s\n" "${w}"
-		fi
-	done
-}
-__helm_compopt() {
-	true # don't do anything. Not supported by bashcompinit in zsh
+	return cmd.GenZshCompletion(out)
 }
-__helm_ltrim_colon_completions()
-{
-	if [[ "$1" == *:* && "$COMP_WORDBREAKS" == *:* ]]; then
-		# Remove colon-word prefix from COMPREPLY items
-		local colon_word=${1%${1##*:}}
-		local i=${#COMPREPLY[*]}
-		while [[ $((--i)) -ge 0 ]]; do
-			COMPREPLY[$i]=${COMPREPLY[$i]#"$colon_word"}
-		done
-	fi
-}
-__helm_get_comp_words_by_ref() {
-	cur="${COMP_WORDS[COMP_CWORD]}"
-	prev="${COMP_WORDS[${COMP_CWORD}-1]}"
-	words=("${COMP_WORDS[@]}")
-	cword=("${COMP_CWORD[@]}")
-}
-__helm_filedir() {
-	local RET OLD_IFS w qw
-	__debug "_filedir $@ cur=$cur"
-	if [[ "$1" = \~* ]]; then
-		# somehow does not work. Maybe, zsh does not call this at all
-		eval echo "$1"
-		return 0
-	fi
-	OLD_IFS="$IFS"
-	IFS=$'\n'
-	if [ "$1" = "-d" ]; then
-		shift
-		RET=( $(compgen -d) )
-	else
-		RET=( $(compgen -f) )
-	fi
-	IFS="$OLD_IFS"
-	IFS="," __debug "RET=${RET[@]} len=${#RET[@]}"
-	for w in ${RET[@]}; do
-		if [[ ! "${w}" = "${cur}"* ]]; then
-			continue
-		fi
-		if eval "[[ \"\${w}\" = *.$1 || -d \"\${w}\" ]]"; then
-			qw="$(__helm_quote "${w}")"
-			if [ -d "${w}" ]; then
-				COMPREPLY+=("${qw}/")
-			else
-				COMPREPLY+=("${qw}")
-			fi
-		fi
-	done
-}
-__helm_quote() {
-	if [[ $1 == \'* || $1 == \"* ]]; then
-		# Leave out first character
-		printf %q "${1:1}"
-	else
-		printf %q "$1"
-	fi
-}
-autoload -U +X bashcompinit && bashcompinit
-# use word boundary patterns for BSD or GNU sed
-LWORD='[[:<:]]'
-RWORD='[[:>:]]'
-if sed --help 2>&1 | grep -q 'GNU\|BusyBox'; then
-	LWORD='\<'
-	RWORD='\>'
-fi
-__helm_convert_bash_to_zsh() {
-	sed \
-	-e 's/declare -F/whence -w/' \
-	-e 's/_get_comp_words_by_ref "\$@"/_get_comp_words_by_ref "\$*"/' \
-	-e 's/local \([a-zA-Z0-9_]*\)=/local \1; \1=/' \
-	-e 's/flags+=("\(--.*\)=")/flags+=("\1"); two_word_flags+=("\1")/' \
-	-e 's/must_have_one_flag+=("\(--.*\)=")/must_have_one_flag+=("\1")/' \
-	-e "s/${LWORD}_filedir${RWORD}/__helm_filedir/g" \
-	-e "s/${LWORD}_get_comp_words_by_ref${RWORD}/__helm_get_comp_words_by_ref/g" \
-	-e "s/${LWORD}__ltrim_colon_completions${RWORD}/__helm_ltrim_colon_completions/g" \
-	-e "s/${LWORD}compgen${RWORD}/__helm_compgen/g" \
-	-e "s/${LWORD}compopt${RWORD}/__helm_compopt/g" \
-	-e "s/${LWORD}declare${RWORD}/builtin declare/g" \
-	-e "s/\\\$(type${RWORD}/\$(__helm_type/g" \
-	-e 's/aliashash\["\(.\{1,\}\)"\]/aliashash[\1]/g' \
-	-e 's/FUNCNAME/funcstack/g' \
-	<<'BASH_COMPLETION_EOF'
-`
-	out.Write([]byte(zshInitialization))
-
-	runCompletionBash(out, cmd)
 
-	zshTail := `
-BASH_COMPLETION_EOF
-}
-__helm_bash_source <(__helm_convert_bash_to_zsh)
-`
-	out.Write([]byte(zshTail))
-	return nil
+func runCompletionFish(out io.Writer, cmd *cobra.Command) error {
+	return cmd.GenFishCompletion(out, true)
```

Comment:
What do you think of making the second parameter configurable to helm's users?
I suggest adding a `--no-descriptions` flag to the `completion` command.
You can get inspiration from the work started here: https://github.com/helm/helm/pull/7690

Reply:
Will try to do that, i was not aware of the work you previously done on this.

Reply:
No problem.  Contributions are always welcomed and help spread the knowledge.

Reply:
should be fixed in c6849c4014c3524bca4b2dd6698594767fa8c386